src/api.js

```diff
@@ -1369,6 +1370,7 @@ api.declare({
       provisionerId, workerType, workerGroup, workerId, count, aborted,
     ),
     this.workerInfo.provisionerSeen(provisionerId),
+    this.WorkerType.create({provisionerId, workerType}, true),
```

Comment:
I think this needs to use the same "seen" logic as provisioners - so something like `this.workerInfo.workerTypeSeen`.